sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java

```diff
@@ -99,13 +113,127 @@ public SentryEvent process(@NotNull SentryEvent event, @NotNull Hint hint) {
+    if (!contentProviderOnCreates.isEmpty()) {
+      final @NotNull SentrySpan contentProviderRootSpan =
+          new SentrySpan(
+              contentProviderOnCreates.get(0).getStartTimestampSecs(),
+              contentProviderOnCreates
+                  .get(contentProviderOnCreates.size() - 1)
+                  .getProjectedStopTimestampSecs(),
+              traceId,
+              new SpanId(),
+              null,
+              UI_LOAD_OP,
+              "Content Providers",
+              SpanStatus.OK,
+              APP_METRICS_ORIGN,
+              new HashMap<>(),
+              null);
+      txn.getSpans().add(contentProviderRootSpan);
+      for (final @NotNull TimeSpan contentProvider : contentProviderOnCreates) {
+        txn.getSpans()
+            .add(
+                timeSpanToSentrySpan(
+                    contentProvider, contentProviderRootSpan.getSpanId(), traceId));
+      }
+    }
+
+    // Activities
+    final @NotNull List<ActivityLifecycleTimeSpan> activityLifecycleTimeSpans =
+        appStartMetrics.getActivityLifecycleTimeSpans();
+    if (!activityLifecycleTimeSpans.isEmpty()) {
+      final SentrySpan activityRootSpan =
+          new SentrySpan(
+              activityLifecycleTimeSpans.get(0).getOnCreate().getStartTimestampSecs(),
+              activityLifecycleTimeSpans
+                  .get(activityLifecycleTimeSpans.size() - 1)
+                  .getOnStart()
+                  .getProjectedStopTimestampSecs(),
+              traceId,
+              new SpanId(),
+              null,
+              UI_LOAD_OP,
+              "Activities",
+              SpanStatus.OK,
+              APP_METRICS_ORIGN,
+              new HashMap<>(),
+              null);
+      txn.getSpans().add(activityRootSpan);
+
+      for (ActivityLifecycleTimeSpan activityTimeSpan : activityLifecycleTimeSpans) {
+        if (activityTimeSpan.getOnCreate().hasStarted()
+            && activityTimeSpan.getOnCreate().hasStopped()) {
+          txn.getSpans()
+              .add(
+                  timeSpanToSentrySpan(
+                      activityTimeSpan.getOnCreate(), activityRootSpan.getSpanId(), traceId));
+        }
+        if (activityTimeSpan.getOnStart().hasStarted()
+            && activityTimeSpan.getOnStart().hasStopped()) {
+          txn.getSpans()
+              .add(
+                  timeSpanToSentrySpan(
+                      activityTimeSpan.getOnStart(), activityRootSpan.getSpanId(), traceId));
+        }
+      }
+    }
+  }
+
+  @NotNull
+  private static SentrySpan timeSpanToSentrySpan(
+      final @NotNull TimeSpan span,
+      final @Nullable SpanId parentSpanId,
+      final @NotNull SentryId traceId) {
+    return new SentrySpan(
+        span.getStartTimestampSecs(),
+        span.getProjectedStopTimestampSecs(),
+        traceId,
+        new SpanId(),
+        parentSpanId,
+        ActivityLifecycleIntegration.UI_LOAD_OP,
+        span.getDescription(),
+        SpanStatus.OK,
```

Comment:
Also, as discussed privately, we could change our auto-instr in a way that it wraps the instrumented methods into try-catch and connects throwables to spans, then we could set the SpanStatus accordingly. And also, if we plan to surface something in the UI, based on the span <-> error connection